internal/services/appservice/helpers/web_app_slot_schema.go

```diff
@@ -815,6 +819,10 @@ func (s *SiteConfigLinuxWebAppSlot) ExpandForUpdate(metadata sdk.ResourceMetaDat
 		expanded.HealthCheckPath = pointer.To(s.HealthCheckPath)
 	}
 
+	if metadata.ResourceData.HasChange("site_config.0.worker_count") {
+		expanded.NumberOfWorkers = pointer.To(int32(s.WorkerCount))
```

Comment:
Similarly here:
```suggestion
		expanded.NumberOfWorkers = pointer.To(s.WorkerCount)
```

Reply:
@jackofallops  thank you for your reply. I've updated the PR with your suggestions.